### PR TITLE
Feature/certification page

### PR DESCRIPTION
**Pull request by brooksideas**
## 📝 Summary

> **Why** is this Pull Request necessary? Provide a concise explanation of the change or feature.  
> For example: "Fixes a bug", "Implements a new feature", or "Bumping package version".

## 📝 Summary

This Pull Request introduces the **Certification Management feature**. It implements several pages and functionalities: Create, Edit, Register, Detail, and List certifications. These additions allow users to manage certification details, register for courses, and view certification information, providing a streamlined experience for both users and admins in managing certifications.

## 📊 Impact

> **Scope**: These changes apply primarily to the certification pages and components, including `CertificationList`, `CertificationDetail`, `CertificationCreateForm`, `CertificationEditForm`, and `CertificationRegister`. Each component and page has specific Vue and Vuex store integrations to handle user interactions, data validation, and persistence.

> **Affected parties**: Only users interacting with certification features are impacted. Admins benefit from streamlined create and edit workflows, while general users can view and register for certifications.

> **Risk Assessment**: Minimal risk, as these changes are contained within the certification feature and do not impact other parts of the system. The workflow is cohesive, starting from the `CertificationList` page (which links to individual `CertificationDetail` pages) and allowing authorized users to navigate to the `CertificationCreate` and `CertificationEdit` pages. Validation and state management are robustly handled, minimizing potential side effects.

## ✅ Testing

> What testing have you done and verified? If you can't test, please say why that is.
> If testing is not possible, explain why and how it will be handled later (manual verification, etc.).
**Manual Testing has been conducted. ** 

## 🔗 References

> **Supporting Information**: Add relevant documentation, discussions, or background for this Pull Request.  
> This could include:
- Links to issue trackers (e.g., GitHub issues)
- API Documentation
- External resources (StackOverflow, Notion pages, third-party package documentation)
**The dayjs package has been added for time keeping which is a great alternative for moment.js. But this might be removed in the future if we do not require a library to handle the time aspect strictly on this project.**
[dayjs](https://www.npmjs.com/package/dayjs)

## The Certification Integration is depicted in the flow chart v1.0 here in the below image.
certification flowchart drawio

### CODE REVIEW

File: src/views/CertificationCreateForm.vue [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+            <!-- Duration -->
+            <FormField label="Duration (Hours)">
+                <div class="flex flex-col gap-y-1.5">
+                    <FormControl v-model="duration" type="number" placeholder="Enter duration in hours" :icon="mdiClock"
+                        :disabled="isSubmitting || isLoading" />
+                    <p v-if="durationError" class="text-red-500">{{ durationError }}</p>
+                </div>
+            </FormField>
+
```
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/views/CertificationCreateForm.vue:65]: dayjs is already installed and will be used in integration task.

File: src/views/CertificationEditForm.vue [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+const certificationId = route.params.id;
+const isLoading = ref(false);
+const generalError = ref('');
+
+// Define validation schema
+const schema = yup.object({
+  title: yup.string().required("Title is required"),
+  description: yup.string().required("Description is required"),
+  rating: yup.number().min(1).max(5).required("Rating must be between 1 and 5"),
+  level: yup.string().required("Level is required"),
```
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/views/CertificationEditForm.vue:146]: `duration: "02:30"` this is the format we expect it , so it is a correct to use a string.

File: src/router/index.js [hunk 3]
```diff
@@ -61,9 +61,33 @@ const routes = [
       title: 'Certification Detail'
     },
     path: '/certifications/:id',
-    name: 'CertificationDetail',
+    name: 'certification-detail',
```
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/router/index.js:64]: We are using snake-case for all route naming.

File: src/views/CertificationCreateForm.vue [hunk 4]
```diff
@@ -0,0 +1,191 @@
+<template>
+    <SectionMain>
+        <SectionTitleLineWithButton :icon="mdiBallotOutline" title="Create Certification" main>
+            <BaseButton label="Back to Certifications" color="contrast" rounded-full small
+                @click="backToCertificationPage" />
+        </SectionTitleLineWithButton>
+        <CardBox is-form @submit.prevent="submit">
+            <div v-if="generalError" class="mb-4 p-4 text-rose-500 bg-rose-300 border border-red-400 rounded">
+                {{ generalError }}
```
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/views/CertificationCreateForm.vue:9]: tailwind is handling it thus, it is not important to create a new central class.

File: src/views/CertificationEditForm.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/views/CertificationEditForm.vue]: fix here 
https://github.com/Ethiopian-Data-Science-Association/edsa-website/pull/11/commits/dde8b2fe0b619d45382512c3b81e43a2b0ae5860

File: src/views/CertificationEditForm.vue [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+const router = useRouter();
+const certificationId = route.params.id;
+const isLoading = ref(false);
+const generalError = ref('');
+
+// Define validation schema
+const schema = yup.object({
+  title: yup.string().required("Title is required"),
+  description: yup.string().required("Description is required"),
+  rating: yup.number().min(1).max(5).required("Rating must be between 1 and 5"),
```
[the remark this answers is not in the document]
Comment (brooksideas) [reply src/views/CertificationEditForm.vue:145]: fix for this comment here.
https://github.com/Ethiopian-Data-Science-Association/edsa-website/pull/11/commits/dde8b2fe0b619d45382512c3b81e43a2b0ae5860

**Review by LibenHailu — CHANGES_REQUESTED** [review]
image

image


There is a contrast issue

**Comment by brooksideas** [comment]
Here is the contract issue fixes. 

image

image

image

**Review by LibenHailu — APPROVED** [review]

**Pull request merged**